modules/videostab/src/frame_source.cpp

```diff
@@ -116,5 +116,15 @@ int VideoFileSource::height() { return ((VideoFileSourceImpl*)impl.get())->heigh
 int VideoFileSource::count() { return ((VideoFileSourceImpl*)impl.get())->count(); }
 double VideoFileSource::fps() { return ((VideoFileSourceImpl*)impl.get())->fps(); }
 
+MaskFrameSource::MaskFrameSource(Ptr<IFrameSource> source): impl(source) {}
+
+void MaskFrameSource::reset() { impl->reset(); }
+Mat MaskFrameSource::nextFrame() {
+    Mat nextFrame = impl->nextFrame();
+    if (!nextFrame.empty())
+        maskCallback_(nextFrame);
```

Comment:
> if (!nextFrame.empty())

This condition is implementation-specific too.
I believe callback should handle this instead:
- callback can generate some pre-defined mask
- or keep passing of the last observed mask